### ISSUE TITLE
license field needs more guidance

### ISSUE THREAD

**Issue by regongithub** [project-open-data/project-open-data.github.io#196]
Some datasets are copyrighted -- can't address that in current field or point to specific license guidance via URL.

**Comment by ajturner** [comment]
I'm interested in continuing this conversation. A URL is a good first step - however it's unclear what should be at the other side of that URL.
 
**As a developer, when enumerating over data, I want to have structured license information that enables me to understand the capabilities this license permits or restricts.**

So can the license point to a JSON such as:

```json
{
  "title": "CC0",
  "version": "1.0",
  "description": "The person who associated a work with this deed has dedicated the work to the public domain by waiving all of his or her rights to the work worldwide under copyright law, including all related and neighboring rights, to the extent allowed by law.

You can copy, modify, distribute and perform the work, even for commercial purposes, all without asking permission.",
  "capabilities": {
    "commercial_use": true,
    "attribution_required": false,
    "allows_redistribution": true,
    "share_modifications": false
  },
  "logo":  "https://creativecommons.org/images/deed/nolaw.png",
  "link": "https://creativecommons.org/publicdomain/zero/1.0/"
}
```

does something like that exist?

**Comment by waldoj** [comment]
Huh. That's such a totally obvious-in-retrospect idea that it seems like it *has* to exist. And yet I've never encountered it.

**Comment by konklone** [comment]
The package manifest in Node-land, `package.json`, [has a license field](https://docs.npmjs.com/files/package.json#license) that will technically accept any text, but asks for ["SPDX" identifiers](https://spdx.org/licenses/). It's the only thing I've seen in the wild like this.

**Comment by philipashlock** [comment]
@ajturner @waldoj @konklone actually @aaronsw's spirit lives on throughout the web in part because he [helped establish the licensing metadata standard](https://www.eff.org/deeplinks/2014/08/aaron-swarts-work-internets-own-boy) for this (as a *teenager* have you) and it's now part of the foundation of Creative Commons that's already widely in use.  @waldoj as I understand it this **is** the CCREL approach you and @JoshData were referring to. Given the era this came from it was originally done as RDF and is now RDFa instead of JSON, but it appears embedded in the pages found with the URLs we use for Creative Commons License Deeds. For example, the [CC-BY Deed](https://creativecommons.org/licenses/by/4.0/) includes the following RDFa:

```html
   <h3 resource="http://creativecommons.org/ns#Reproduction"
        rel="cc:permits">You are free to:</h3>
    <ul class="license-properties">
      <li class="license share"
          rel="cc:permits"
          resource="http://creativecommons.org/ns#Distribution">
        <strong>Share</strong>  &mdash; copy and redistribute the material in any medium or format
      </li>
        <li class="license remix"
            rel="cc:permits"
            resource="http://creativecommons.org/ns#DerivativeWorks">
          <strong>Adapt</strong>  &mdash; remix, transform, and build upon the material
        </li>
        <li class="license commercial">
          for any purpose, even commercially.
        </li>
      <li id="more-container"
          class="license-hidden">
        <span id="devnations-container" />
      </li>
    </ul>
```

That said, it would be nice to not have to parse the RDFa out of the HTML here - and not to even have to parse the HTML in the first place. Within the HTML there is also a link to the raw RDF XML of the deed:

```html
<link rel="alternate" type="application/rdf+xml" href="rdf" />
```
In context, that href resolves to https://creativecommons.org/licenses/by/4.0/rdf but ideally this would be provided in alternate formats including JSON or JSON-LD and ideally could be requested directly with HTTP using [Content Negotiation](http://en.wikipedia.org/wiki/Content_negotiation) based on an `Accept` header rather than parsed out of HTML, i.e. I'd want the response from `curl -I https://creativecommons.org/licenses/by/4.0/` to include: 

`Accept: application/rdf+xml, application/ld+json`

As @JoshData said, I think this is still pretty obscure and non-obvious, so guidance would be needed, but in practice I think most of these URLs will be used more as commonly understood unique identifiers and human readable landing pages rather than a bespoke set of conditions that need to be parsed. 

Of course, that thought implies that we actually would have a canonical URL for something like "U.S. Public Domain" At this point, I think we should have something like that (in addition to guidance on using alternatives like CC0 for expanding use internationally) and that it should function in a very similar manner as the [Creative Commons Public Domain Mark](http://creativecommons.org/publicdomain/mark/1.0/), but be specific and limited to the U.S. jurisdiction as is the case with U.S. Government Works ([Title 17 § 105](http://www.copyright.gov/title17/92chap1.html#105)) and other reasons like [Title 37: §202.1](http://copyright.gov/title37/202/37cfr202-1.html) or expired copyrights. This should have a simple human readable page like Creative Commons deeds, include machine readable rights, and also link to the full legal documentation. 

I don't think this URL should live on the Project Open Data site since this isn't specific to data, but as we look to establish this somewhere we may need to provide an interim URL such as http://www.usa.gov/copyright.shtml or something on project-open-data.cio.gov

I'm cc'ing @peterspdx here in reference to https://github.com/creativecommons/Localized-Public-Domain-Mark

**Comment by waldoj** [comment]
Well, this is delightful. :) Thanks for a very informative and engaging explanation, @philipashlock!

**Comment by konklone** [comment]
Wow. A whole bunch of great stuff there, @philipashlock. I had no idea about license metadata, or what Creative Commons' plans for a version 2 are.

Also https://github.com/creativecommons/Localized-Public-Domain-Mark looks like exactly what we'd want (though it's new and not finalized)?

cc @tvol @JoshData

**Comment by tvol** [comment]
@konklone @philipashlock @JoshData yeah I believe that's in more of an idea stage right now. Diane Peters (@peterspdx) is on point.

**Comment by rebeccawilliams** [comment]
ICYMI, I wanted to flag the recent **[licensing guidance updates](https://project-open-data.cio.gov/open-licenses/)**, created with #454 + #456. 

You'll notice that a new section for [U.S. Government works](https://project-open-data.cio.gov/open-licenses/#us-government-works) has been added, highlighting the importance of a worldwide public domain dedication (CC-0) and providing a default URL for U.S. Government works: `http://www.usa.gov/publicdomain/label/1.0/`

Interestingly, I learned *today* that the White House's Flickr page has separately and of their own accord been referencing the same U.S. Government Works page, [see this example](https://www.flickr.com/photos/whitehouse/16424072323/). Great minds point to a common temporary solution! 

The content of the U.S. Government Works page is a work in progress and will ideally incorporate a Creative Commons' solution to a U.S. specific version of the Public Domain Mark or to an equivalent legal/human-readable/machine-readable solution.

**Comment by akuckartz** [comment]
One minor point: RDFa *is* an RDF serialization. But providing a JSON-LD alternative would help.

Maybe someone can add information on the CC Wiki. There is a page on [RDFa](https://wiki.creativecommons.org/Rdfa) and [RDF/XML material is available](https://github.com/creativecommons/license.rdf/tree/master/cc/licenserdf/licenses) but no info about JSON-LD.

**Comment by prototypo** [comment]
@philipashlock, I would like to expound a bit on @akuckartz's comment regarding RDF. You said, "originally done as RDF and is now RDFa instead of JSON", and @akuckartz rightly pointed out that RDFa is an RDF serialization.

RDF is a format-independent data model. There are, as of last year, 7 standardized syntaxes for that data model, including a JSON one known as JSON-LD. Please see http://json-ld.org/ for more information, or http://www.w3.org/TR/json-ld/ for the standard.

A more accurate way to phrase your statement would have been something like, "was written in the RDF data model and serialized as RDF/XML. We all know that RDF/XML sucked, so is now available in other standard RDF formats, including RDFa, which is easily convertible using widely available tooling to JSON-LD. That gets us where we need to go." :-)

There is **no reason** to jettison the work done in describing or representing licenses on the Web because you want to use JSON. JSON-LD is JSON. That allows you to make use of the Creative Commons work, and still integrate easily into existing Web tools.

Please don't reinvent the wheel on licensing for the Web. The CC folks have put a lot of very valuable thought into this, as have other parts of the Web community.
